signature/jws/envelope_test.go

```diff
@@ -38,65 +141,436 @@ func getSignReq(signingScheme signature.SigningScheme) (*signature.SignRequest,
 		Signer:                   signer,
 		SigningTime:              time.Now(),
 		Expiry:                   time.Now().Add(time.Hour),
-		ExtendedSignedAttributes: nil,
+		ExtendedSignedAttributes: extendedSignedAttribute,
 		SigningAgent:             "Notation/1.0.0",
 		SigningScheme:            signingScheme,
 	}, nil
+}
+
+func getSigner(isLocal bool, certs []*x509.Certificate, privateKey *rsa.PrivateKey) (signature.Signer, error) {
+	if certs == nil {
+		certs = getSigningCerts()
+	}
+	if privateKey == nil {
+		privateKey = testhelper.GetRSALeafCertificate().PrivateKey
+	}
+	if isLocal {
+		return signature.NewLocalSigner(certs, privateKey)
+	}
 
+	return &remoteSignerMock{
+		certs:      certs,
+		privateKey: privateKey,
+	}, nil
```

Comment:
For test signer, should we also cover *ecdsa.PrivateKey and different key sizes for each keyType?

Reply:
added.
>--- PASS: TestSignVerify (1.05s)
    --- PASS: TestSignVerify/RSA_2048 (0.13s)
    --- PASS: TestSignVerify/RSA_3072 (0.31s)
    --- PASS: TestSignVerify/RSA_4096 (0.56s)
    --- PASS: TestSignVerify/ECDSA_256 (0.01s)
    --- PASS: TestSignVerify/ECDSA_384 (0.01s)
    --- PASS: TestSignVerify/ECDSA_521 (0.02s)